relay-kafka/src/lib.rs

```diff
@@ -0,0 +1,54 @@
+pub mod config;
+#[cfg(feature = "processing")]
+mod producer;
+#[cfg(feature = "processing")]
+mod utils;
+
+use failure::Fail;
+
+#[cfg(feature = "processing")]
+pub use producer::*;
```

Comment:
Since we already create a config module, we could also make the producer module public and bundle all the types there.

There's not yet a consistent paradigm for how to structure such crates in Relay. I've mostly used this heuristic:
 - If it's something that you almost always import when using a crate/module, export it top-level.
 - If it's something you rarely use or do not need to import directly (e.g. because it's a return type), move it into submodules.
 - If there's several concerns that would get messy in a single namespace, put everything behind modules.